microsoft/iot-cardboard-js
# Integrating Advanced search to Element form

### Summary of changes 🔍 
This PR integrates Advanced Search functionality with the Element form. This new feature allows the users to select a Primary twin after building a query to search for it. Previously the Element form only allowed users to select a Twin by searching for its `$dtId`, now both `$dtId` and advanced search are enabled.

Also included in this PR a button was added to the Element form to show a callout containing the `PropertyInspector`. The `PropertyInspector` component here will allow users to know which properties are part of the selected Twin in case they want to search for another Twin of the same model. 

### Testing 🧪
PropertyInspector:
- Navigate to Element form
- If a Twin is selected: (else select Twin from list)
  - Click the first icon besides the Primary Twin dropdown.
  - Make sure properties are correct for the selected Twin model

Advanced search:
- Navigate to Element form
- Click the second icon besides the Primary Twin dropdown
- Create a query to search for Twins, in Mock data any Operator other than Equals will return all available data
  - An example of an Equals operation that will yield some filtered values is: 
![image](https://user-images.githubusercontent.com/55852250/189770490-d7899dbf-720c-40f4-b272-2172d39cd248.png)

- Select a twin from the list
- Click on the `Select` button in the modal
- Review:
  - If dropdown changed to selected twin
  - If Property inspector shows correct twin data
  - If saving the form persists data


### Checklist ✔️
